src/components/ui/input/index.js

```diff
@@ -0,0 +1,69 @@
+import React, {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import './input.css';
+
+const MIN_HEIGHT = 20;
+
+const Input = ({
+  characterLimit = false,
+  defaultValue = '',
+  size = 'large',
+}) => {
+  const isLimitSet = characterLimit > 0;
+  const ref = useRef(null);
+  const [value, setValue] = useState(defaultValue);
+
+  useEffect(() => {
+    setValue(defaultValue);
+  }, [defaultValue]);
+
+  useLayoutEffect(() => {
+    ref.current.style.height = 'inherit';
+
+    ref.current.style.height = `${Math.max(
+      ref.current.scrollHeight,
+      MIN_HEIGHT
+    )}px`;
+  }, [value]);
+
+  const handleChange = (e) => {
+    setLimitedLengthText(e.target.value);
+  };
+
+  const setLimitedLengthText = useCallback(
+    (text) => {
+      isLimitSet
+        ? setValue(text.slice(0, characterLimit))
+        : setValue(text.slice(0));
+    },
```

Comment:
following my comment above, if `characterLimit` is set to 0 by default then we could simplify this to 
```suggestion
    (text) => setValue(text.slice(0, characterLimit));
```

Reply:
I see what you're saying but I don't think this is going to work. If slice is set to `slice(0, characterLimit)` when the `characterLimit = 0`, then the method [starts and ends at 0](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice), meaning nothing is returned. So I believe we do need this, the check above, and the ability to set the default value to `false`.